### PR TITLE
New module: iptables_raw

### PR DESCRIPTION
**Pull request by tgates81**
##### ISSUE TYPE

 - Feature Pull Request
 - New Module Pull Request
 - Bugfix Pull Request
 - Docs Pull Request

##### ANSIBLE VERSION

```

```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by tgates81** [comment]
Any interest in this module? No human activity for a long time.

File: lib/ansible/modules/system/iptables_raw.py [hunk 1]
```diff
@@ -0,0 +1,556 @@
...
+                    'supported_by': 'community',
+                    'metadata_version': '1.1'}
+
+DOCUMENTATION = """
+---
+module: iptables_raw
+short_description: Modify runtime iptables rules
+requirements: []
+version_added: "2.7"
+author: Tyler Gates <tgates81@gmail.com>
```
Comment (gundalow) [inline lib/ansible/modules/system/iptables_raw.py:33]:
```suggestion
author:
  - Tyler Gates (@tgates81) <tgates81@gmail.com>
```

File: lib/ansible/modules/system/iptables_raw.py [hunk 2]
```diff
@@ -0,0 +1,556 @@
...
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'metadata_version': '1.1'}
+
+DOCUMENTATION = """
+---
+module: iptables_raw
+short_description: Modify runtime iptables rules
+requirements: []
```
Comment (gundalow) [inline lib/ansible/modules/system/iptables_raw.py:31]:
```suggestion
```

File: .github/BOTMETA.yml [hunk 3]
```diff
@@ -316,6 +316,7 @@ files:
   $modules/system/gconftool2.py: Akasurde
   $modules/system/group.py: $team_ansible
   $modules/system/interfaces_file.py: obourdon
+  $modules/system/iptables_raw.py: tgates81
```
Comment (gundalow) [inline .github/BOTMETA.yml:319]: This is only needed when `author` isn't set in the module, so the change to this file can be reverted
```suggestion
```

File: lib/ansible/modules/system/iptables_raw.py [hunk 4]
```diff
@@ -0,0 +1,556 @@
...
+version_added: "2.7"
```
Comment (gundalow) [inline lib/ansible/modules/system/iptables_raw.py:32]:
```suggestion
version_added: '2.8'
```

File: lib/ansible/modules/system/iptables_raw.py [hunk 5]
```diff
@@ -0,0 +1,556 @@
...
+  - chain_* and rule_* options require a list of dictionary values. The key
+    should designate the task number and the value will be the rule or chain.
+options:
+  ipv:
+    description:
+      - Use protocol 4 or 6 when creating rules.
+    required: false
+    default: "4"
+    choices: [ "4", "6" ]
+    version_added: "2.7"
```
Comment (gundalow) [inline lib/ansible/modules/system/iptables_raw.py:53]: As this is a new module, we only need the top level `version_added`, no need for it per option.
```suggestion

```

File: lib/ansible/modules/system/iptables_raw.py [hunk 6]
```diff
@@ -0,0 +1,556 @@
...
+    mid-flight lockout induced by multiple module calls while building the
+    rules.
+notes:
+  - chain_* and rule_* options require a list of dictionary values. The key
+    should designate the task number and the value will be the rule or chain.
+options:
+  ipv:
+    description:
+      - Use protocol 4 or 6 when creating rules.
+    required: false
```
Comment (gundalow) [inline lib/ansible/modules/system/iptables_raw.py:50]: This is the default, so can be removed
```suggestion
```

File: lib/ansible/modules/system/iptables_raw.py [hunk 7]
```diff
@@ -0,0 +1,556 @@
...
+# 5) Set default policy for INPUT to 'ACCEPT'.
+# 6) Append -j DROP to INPUT.
+# 7) Insert tcp port 22 rule to INPUT.
+# 8) Remove tcp port 21 rule from INPUT.
+# 9) Flush the 'TESTING' table.
+# 10) Remove 'OLD_CHAIN' chain.
+# 11) Rename chain 'SRC_CHAIN' to 'DST_CHAIN'.
+# 12) Append -j MASQUERADE for eth0 on the POSTROUTING chain on the nat table.
+
+- iptables_raw:
```
Comment (gundalow) [inline lib/ansible/modules/system/iptables_raw.py:147]: Please set a `name:` for all EXAMPLES
```suggestion
- name: Example
  iptables_raw:
```

File: lib/ansible/modules/system/iptables_raw.py [hunk 8]
```diff
@@ -0,0 +1,556 @@
...
+            required_one_of=[
+                [
+                    "rules_append", "rules_insert", "rules_absent",
+                    "chains_present", "chains_absent", "chains_policy",
+                    "chains_rename", "chains_flush"
+                ],
+            ],
+            argument_spec=dict(
+                ipv=dict(
+                    required=False,
```
Comment (gundalow) [inline lib/ansible/modules/system/iptables_raw.py:222]:
```suggestion
```

File: lib/ansible/modules/system/iptables_raw.py [hunk 9]
```diff
@@ -0,0 +1,556 @@
...
+                    default="4",
+                    choices=["4", "6"],
+                    type="str",
```
Comment (gundalow) [inline lib/ansible/modules/system/iptables_raw.py:225]: Should this be int, rather than string?

**Comment by mkrizek** [comment]
Thank you for your interest in Ansible! We are no longer accepting new plugins to ansible/ansible directly. Please re-file this pull request against https://www.github.com/ansible-collections/community.general.

If you have further questions please stop by IRC or the mailing list:
* IRC: #ansible on irc.freenode.net
* mailing list: https://groups.google.com/forum/#!forum/ansible-project

**Pull request closed without merging**